crates/libs/registry/src/key.rs

```diff
@@ -98,6 +98,15 @@ impl Key {
         unsafe { self.set_value(name, REG_SZ, value.as_ptr() as _, value.len() * 2) }
     }
 
+    /// Sets the name and value in the registry key.
+    pub fn set_hstring<T: AsRef<str>>(
+        &self,
+        name: T,
+        value: &windows_strings::HSTRING,
+    ) -> Result<()> {
+        unsafe { self.set_value(name, REG_SZ, value.as_ptr() as _, value.len() * 2) }
+    }
```

Comment:
I am fine with just providing this method but it does feel like there's a version of this that could be more generic, in a similar way to `AsRef<str>`. We don't need an `&HSTRING` per-se just something that provides an `&[u16]` that's guaranteed to be null-terminated.

My only hesitation with that approach is it does increase complexity. It'd probably need a new type or `unsafe` trait to express the validity requirement.

Reply:
I was thinking the same thing. In this case it doesn't have to be null terminated, but in general yes that's the problem and why I didn't want to use `PCWSTR` and thus make the method `unsafe`.

Reply:
The `HSTRING` just seems like a reasonable compromise because it does guarantee the right memory layout and is safe and nicely convertible. Perhaps this is where we can pick up @riverar's suggestion and provide a `set_typed_value` kind of method that takes a slice and provides the added flexibility and micro-optimization if really needed.

Reply:
Ah right, so it could just be `AsRef<[u16]>`? Though that would imply needing an `AsRef` implementation for `HSTRING`.

Reply:
Oh sorry, missed your second comment. That does convince me it's not really needed.